github/open_close_pr.py

```diff
@@ -0,0 +1,184 @@
+#!/usr/bin/python3
+
+import os
+import github3
+import argparse
+import logging
+import yaml
+from cachecontrol.adapter import CacheControlAdapter
+import git
+from git import Repo
+
+
+REF_FORMAT = 'refs/heads/'
+DEFAULT_COMMIT_MSG = 'automated commit'
+FREEIPA_PRCI_CONFIG_FILE = '.freeipa-pr-ci.yaml'
+REMOTE_REPO = 'https://github.com/freeipa/freeipa.git'
+UPSTREAM_REMOTE_REF = 'upstream'
+MYGITHUB_REMOTE_REF = 'mygithub'
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+consoleHandler = logging.StreamHandler()
+consoleHandler.setLevel(logging.INFO)
+logger.addHandler(consoleHandler)
+
+
+def load_yaml(yml_path):
+    try:
+        with open(yml_path) as yml_file:
+            return yaml.load(yml_file)
+    except IOError as exc:
+        raise argparse.ArgumentTypeError(
+            'Failed to open {}: {}'.format(yml_path, exc))
+    except yaml.YAMLError as exc:
+        raise argparse.ArgumentTypeError(
+            'Failed to parse YAML from {}: {}'.format(yml_path, exc))
+
+
+class AutomatedPR(object):
+
+    def __init__(self, github_token, repo):
+        github = github3.login(token=github_token)
+        github.session.mount('https://api.github.com', CacheControlAdapter())
+        self.repo = github.repository(repo['owner'], repo['name'])
+
+    def commit_new_prci_config_file(self, new_prci_config, args):
+        """
+        Updates the .freeipa-pr-ci.yaml file with the content
+        of the --prci_config provided file.
+        """
+        prci_config_file = os.path.join(args.repo_path,
+                                        FREEIPA_PRCI_CONFIG_FILE)
+
+        content = ''
+        with open(new_prci_config, 'r') as f:
+            content = f.read()
+
+        with open(prci_config_file, 'w') as f:
+            f.write(content)
```

Comment:
I think using [`shutil.copy`](https://docs.python.org/3/library/shutil.html#shutil.copy) is a cleaner solution.